src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,959 @@
+const MUSTARD = '#ffdb58'
+const GRAY = '#808080'
+
+const DEFAULT_COLORS = [
+    RED,
+    ORANGE,
+    YELLOW,
+    GREEN,
+    BLUE,
+    PURPLE,
+    CYAN,
+    MAGENTA,
+    VERDANT,
+    VIOLET,
+    MUSTARD,
+    GRAY,
+]
+
+const RAINBOW = [
+    '#FF0000',
+    '#FF6700',
+    '#FFE000',
+    '#51FF00',
+    '#00F4E8',
+    '#00B1FF',
+    '#0055FF',
+    '#C000FF',
+    '#FF00F9',
+    '#FF0080',
+]
+
+const MAXIMUM_ALLOWED_PROPERTIES = 12
+
+enum Preset {
+    Custom,
+    Primes,
+    Abundant_Numbers,
+    Abundant_Numbers_And_Primes,
+    Polygonal_Numbers,
+    Rainbow_Colored_By_Last_Digit,
+    Rainbow_Colored_By_Last_Digit_Alternative,
+}
+
+enum PathType {
+    Spiral,
+    Rows,
+    Rows_Reset_Augmented_By_One,
+    Rows_Reset_Augmented_By_One_And_Values_Are_Squared,
+}
+
+enum Direction {
+    Right,
+    Left,
+    Up,
+    Down,
+    StartNewRow,
+    None,
+}
+
+enum Property {
+    None,
+    Number_Sequence,
+    Prime,
+    Negative,
+    Even,
+    Divisible_By_Three,
+    Divisible_By_Four,
+    Divisible_By_Five,
+    Divisible_By_Six,
+    Divisible_By_Seven,
+    Divisible_By_Eight,
+    Ends_With_Zero,
+    Ends_With_One,
+    Ends_With_Two,
+    Ends_With_Three,
+    Ends_With_Four,
+    Ends_With_Five,
+    Ends_With_Six,
+    Ends_With_Seven,
+    Ends_With_Eight,
+    Ends_With_Nine,
+    Triangular_Number,
+    Square_Number,
+    Pentagonal_Number,
+    Hexagonal_Number,
+    Heptagonal_Number,
+    Octagonal_Number,
+    Sum_Of_Two_Squares,
+    Abundant,
+    Perfect,
+    Deficient,
+    Semi_Prime,
+}
+
+enum PropertyVisualization {
+    Primary_Color,
+    Secondary_Color,
+}
+
+function getProperty(
+    value: Property,
+    index: number,
+    hasVisibleDependencyAndPredicate: boolean
+) {
+    const property = {
+        value: value,
+        from: Property,
+        displayName: `Property ${index + 1}`,
+        required: false,
+        visibleDependency: '',
+        visiblePredicate: (dependentValue: Property) =>
+            dependentValue !== Property.None,
+    }
+
+    if (hasVisibleDependencyAndPredicate) {
+        property.visibleDependency = `property${index - 1}`
+        property.visiblePredicate = (dependentValue: Property) =>
+            dependentValue !== Property.None
+    }
+
+    return property
+}
+
+function getPropertyVisualization(
+    value: PropertyVisualization,
+    index: number
+) {
+    const propertyVisualization = {
+        value: value,
+        from: PropertyVisualization,
+        displayName: `Property ${index + 1} Visualization`,
+        required: false,
+        visibleDependency: `property${index}`,
+        visiblePredicate: (dependentValue: Property) =>
+            dependentValue !== Property.None,
+    }
+
+    return propertyVisualization
+}
+
+function getPropertyColor(value: string, index: number) {
+    const propertyColor = {
+        value: value,
+        forceType: 'color',
+        displayName: `Property ${index + 1} color:`,
```

Comment:
Kate: remove colon

See https://github.com/numberscope/frontscope/pull/164#discussion_r1046404099 for the original comment.

Reply:
Should be done in https://github.com/numberscope/frontscope/pull/230/commits/b8fdb093b701217c19d8a13a7886748bafdcf132. Someone please check and resolve this conversation if so.

Reply:
looks good.